cmd/download.go

```diff
@@ -0,0 +1,250 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/samply/blazectl/fhir"
+	"github.com/samply/blazectl/util"
+	"github.com/spf13/cobra"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"strings"
+	"time"
+)
+
+var resourceType string
+var outputFile string
+var fhirSearchQuery string
+
+type downloadStats struct {
+	totalPages                            int
+	resourcesPerPage                      []int
+	requestDurations, processingDurations []float64
+	totalBytesIn                          int64
+	startTime                             time.Time
+	endTime                               time.Time
+}
+
+func (ds *downloadStats) String() string {
+
+	builder := strings.Builder{}
+	builder.WriteString(fmt.Sprintf("Pages		[total]			%d\n", ds.totalPages))
+
+	var resourcesTotal int
+	for _, res := range ds.resourcesPerPage {
+		resourcesTotal += res
+	}
+	builder.WriteString(fmt.Sprintf("Resources 	[total]			%d\n", resourcesTotal))
+
+	if len(ds.resourcesPerPage) > 0 {
+		sort.Ints(ds.resourcesPerPage)
+		var totalResources int
+		for _, v := range ds.resourcesPerPage {
+			totalResources += v
+		}
+
+		builder.WriteString(fmt.Sprintf("Resources/Page	[min, avg, max]		%d, %d, %d\n", ds.resourcesPerPage[0], totalResources/len(ds.resourcesPerPage), ds.resourcesPerPage[len(ds.resourcesPerPage)-1]))
+	}
+
+	builder.WriteString(fmt.Sprintf("Duration	[total]			%s\n", ds.endTime.Sub(ds.startTime).Round(time.Second)))
```

Comment:
please output millis, if under one second and seconds with one decimal digit if over one second

Reply:
Have been testing different approaches to tackle this issue (humanize libs, completely custom format function etc.). In the end I came to the conclusion that a simple additional format function for the given context is sufficient. The function would be able to fall back to how duration instances are formatted if it was outside of the range [1s, 1m). For a duration <1s the formatting is done with a millisecond precision.

Due to the fallback we are forced to keep it consistent with the existing formatting. Thus, 1s12ms would be appropriate rather than 1.012s.

Reply:
You are right. Please use 1s12ms. Please implement also the minute step. Hours are not needed.